### DISCUSSION TITLE
2024-07-29 회의록

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#119] [회의록]
### 🔥 회의 기록 관련

- 본인이 한 말들 코멘트로 기록하기

### 🔥 참여 기업 튜터 피드백 건

- 오늘 오후 2시 참여기업 튜터 피드백이 예정되어 있어 readme 초안을 작성하였습니다. 그 간 작성된 문서를 기반으로 작성하였는데 각자 검토해주시고 양질의 피드백을 위해 추가하면 좋을 내용을 말씀해주시면 내용 추가하겠습니다! [링크](https://github.com/nbc-final-a10/TripBuddy)
- (예) 프로젝트 핵심 기술, 서비스의 핵심 기능, 피드백 받고 싶은 내용(코드, 아키텍쳐 등)

  - **[아키텍처]** : atoms, molecules, organisms 패턴 관련, 잘 사용하고 있는건지?
  - **[컴포넌트재사용]** : 어디까지 재사용 가능하게 만들어야 하는지? 
  - **[인증/인가]** : auth.context.tsx 에서 부분적으로 tanstack query 사용하는 문제
  - **[인증/인가]** : 네이버 로그인 (편법) 구현 관련
  - **[pwa]** : 이렇게 하는게 과연 좋은지? ==> 카메라라던가 그런 네이티브에 가까운 기능에 접근?
  - **[BE]** : 프로덕션 레벨에서 Supabase라는 서비스의 사용 빈도가 높은지? 아니라면 FE 엔지니어가 우선 학습해야 하는 서비스는?

==> 참여기업 튜터 피드백에서 답변 받지 못한 관계로, 정리하여 담당 튜터님께 문의 예정

### 🔥 디자이너님 전달 사항 결정

**- 검색 페이지 변동 사항**
  - 현재 와이어프레임에 나와 있는 모바일, PC 버전은 검색 페이지의 작성 로직이 달라 공수가 클 것으로 보임 -> 모바일, PC 환경 둘 다 1페이지 내에서 검색과 
  - 모바일 검색은 위치, 날짜 선택은 별도의 컴포넌트로 이동해서 검색 데이터를 사용자에게 입력 받아야 하는데, 이 입력 받은 데이터를 호출한 페이지에 넘겨 받는 방식이 되어야 하는데, 검색 결과 페이지에서 상단에 계속 노출되어 있는 검색 버튼은 별도 페이지로 이동하지 않고 검색 결과 페이지 내에서 검색하게 되면 로직을 두 번 구현해야 하기 때문에 상당한 공수가 예상됨.
  - -> 구현은 가능하나, 상당한 공수가 들어갈 것으로 판단되어
  - -> 1페이지 내에서 아코디언으로 접는 방식으로 구현하는 것이 가장 현실적으로 보임. -> 이 방식이 UI/UX 관점에서 좋지 않다고 한다면 MVP 발표 이후 2주 간 도전 과제로 넘기는 것이 좋을 것 같음. (오은, 김병준, 곽현정 논의 결과)
  
  - 검색 결과 페이지 pc : 가로 스크롤을 모두 그리드로 바꾸고, 처음엔 두 줄만 띄워주고 '더보기'를 누르면 줄이 추가되는 거로 변경하면 좋을 것 같습니다.
  - trip 카드가 안에 내용은 거의 똑같은데 ui가 다른 버전으로 세 개가 있는데, 구분하지 않아도 된다면 통일하는 게 어떨까요? 안에 들어가는 내용이 trip 목록 페이지에서 더 풍부해진다면 구분하는 게 맞지만, 현재 비교해보니 비슷한 것 같아서 말씀 드려봅니다!
  - trip 카드의 버튼 **[참여하기]** / **[저장하기] [채팅하기]** & trip 상세 페이지의 버튼 **[찜하기] [참가하기]** 이름을 통일하면 좋을 것 같습니다.
    - 왼쪽 버튼은 **찜하기**와 **저장하기** 두 단어 중 하나를 골라서 전부 통일 필요
    - 오른쪽 버튼은 유저가 해당 trip의 참여자일 경우 [채팅하기]로, 참여자가 아닐 경우 [참여하기]로 통일하는 건 어떨까요!
  - trip에서 **[나가기]** 버튼은 상세 페이지에 버튼 영역 하나를 추가해서 넣으면 좋을 것 같다.. 고 회의 때 말씀 드렸었는데요... 지금 댓글 작성하면서 생각난 건데 **내 프로필 페이지에서만 나가기 버튼 누를 수 있도록 하자고** 전에 얘기 나눴던 것 같기도 합니다.

### 🔥 디자이너님 전달 사항 결정

- Chip 지역선택 노랗게 안 되는 부분
- 3명까지 선택가능 << 이 부분도 옵션으로 -> 유지보수성 향상

### 🔥 엔지니어 별 안건

- **[유상]** .env.local 파일에 토스 테스트 결제 key 삭제 요청
- **[은]**
  - 온보딩 일부 페이지 (현 와이어프레임 chip 5개 -> 3개로) 구현 후 MVP 및 UT 배포
  - 모집중인 여정 전체보기 페이지 와이어프레임에 검색창이 추가된 부분 디자이너님과 추가논의 예정
- **[현정]**
  - 버디 chips, 여정 chips '최대 3개까지...' 라벨도 옵션으로 변경 요청 (To. 은)
  - 검색 페이지도 라벨 옵션 설정 요청 (To.병준)
- **[병준]**
  - 함수형 컴포넌트 정의 컨벤션 재확인 (표현식)
  - params 받는 방법 컨벤션으로 정해야 하는지 여부 -> (은) 서버 컴포넌트냐, 클라이언트 컴포넌트냐에 따라서 용법이 달라지기 때문에 컨벤션으로 정하는 게 아니라 용법에 따라 다르게 정하면 됨.

### 🔥 참여 기업 튜터 피드백 내용 공유 및 검색 조건 축소 찬반에 대한 안건

```
2차 참여기업 서면 피드백

1. PWA에 리얼타임, 브로드캐스트까지 적절히 사용하려는 것으로 보이는데 이 부분을 제대로 구현하실지 기대가 됩니다.
온전한 하나의 프러덕트를 만들려고 하시는 의지가 보이는 기술 스펙들로 보여집니다.
PWA까지 접목시켜 구체적인 기술 스택을 채택한 것이 적절해 보입니다.

2. 트래블월렛에 최근 여행 메이트 찾기, 여행 소셜 네트워크 탭이 등장했습니다.
이 프로젝트를 보고 그 비즈니스가 떠올랐는데,
여전히 여행객들이 많고 그 안에서 구현할 수 있는 재미있는 비즈니스인 것 같아서
현재 시장에 놓여 있는 기회를 충분히 잡을 수 있는 비즈니스 프로젝트인 것 같습니다. 항상 필요하고 아직도 요구가 많은 비즈니스라고 생각됩니다.

3. 무얼 하려는지 명확히 보이는 것 같아 기대가 됩니다. 해결하려는 목표가 잘 보입니다.

4. 디자인이 일단 심플하고 직관적으로 눈에 들어오는 것 같아서 기대가 되는 프로젝트입니다.
다만 디자인 요소들을 보아하니 재사용성을 잘 고려해서 코드를 작성하면 낭비를 줄일 수 있을 것 같다는 생각이 들었습니다.
이 부분 참고해주시면 좋을 것 같습니다. 아직 미비해 보이지만 와이어프레임부터 적절해 보입니다.
단, 검색 화면에서는 너무 많은 정보를 한 번에 입력하려는 것처럼 보입니다."
```

**생각해볼 점**

- 우리 서비스의 목적(비즈니스 로직)이 무엇인가? => 나에게 꼭 맞는 여정, 버디 추천
- 그렇다면 검색 로직을 줄이는 게 맞는가?
- 사용자 관점에서 너무 많은 UI를 제공하고 있나?
- 기술 구현이 기간 내 가능한가?

**이강민 튜터님 상세 피드백**

- UI/UX 관점에서 전혀 과하지 않은 옵션으로 보임
- 개발 난이도 상에서도 과하지 않을 것으로 보임
- 프로젝트의 목적을 생각했을 때 '나에게 꼭 맞는 여정/버디 추천 서비스'라면 현행 유지해도 좋을 듯

#### 다른 팀에 비해서 우리 조가 잘 되고 있는 점

- 비즈니스 문제를 해결하려는 노력이 기획 단계에서 보임. (여행 친구 모집...)
- 구현을 위한 기술 스택, 라이브러리, 아키텍쳐의 설정이 구체적이고 현실적임.
- 서비스를 위한 기술 스택 채용이 적절함. (리얼타임 등)
- 서비스를 위한 기술 스택 채용이 적절함과 동시에 그 기술 스택을 왜 채택했는지, 그 기술 스택으로 어떠한 비즈니스 문제를 해결하고자 하는 지가 잘 기획됨. 예를 들어 리얼타임 기능으로 채팅 기능을 만들어 해당 여행에 참여한 멤버들이 별도의 플랫폼을 이용하지 않고 자사의 플랫폼 내에서 실시간 소통이 이루어지게 한다, 스토리 기능을 탑재하여 부족한 커뮤니티의 기능을 보완하고 여행과 관련된 스토리를 유저들이 직접 업로드하면서 더미 데이터에 대한 공수를 서비스 제공자에게서 줄이고, 유저들 스스로 바이럴 마케팅 효과를 볼 수 있도록 함. ===>>> 하지만 이 부분은 MVP 이전까지 더 상세하게 명시해야 할 것으로 생각함.
  - 이 부분에서 우리 프로젝트는 '어떻게' 서비스를 운영할 것인가라는 서비스 엣지가 잘 설명이 됨.

#### 다른 팀의 피드백 내용을 보고 더 생각해볼만한 점

- 정확한 시장 분석 부족
- 서비스 출시 후 기대효과 설명 부족 (정확한 수치)
  - 프로젝트 자체에 의의를 두고 있는지, 비즈니스 문제를 해결하기 위한 서비스 기획인지 명확하게. (후자까지 고려해야 베스트)

### 🔥 개인 분담 티켓별 일정 확인

- 아래 내용을 개인 별로 MVP 발표 이전까지 Projects 보드를 기준으로 체크합니다!
  - 하고 있는 작업
  - 예상 마감 기한
  - 남은 작업
  - MVP 발표 이전까지(8/1 목) 구현 완료 가능 여부 / 트러블 슈팅(기술적 어려움)
  - MVP 발표 이전 (임시로) 축소해야 할 UI 또는 기능
    - 어려움 느끼셔도 괜찮습니다~ 현실적으로 판단해서 편하게 말씀해주세요!

### 🔥 지원금 담당자 지정 건

**- 팀원 현황**
  - 오은 : 국취제
  - 김민지 : 국취제
  - 곽현정 : 국취제
  - 김병준 : 실업급여
  - 최유상 : 해당없음
- 국취제도 기타소득 발생 시 문제가 되는지, 자세한 내용은 행정팀에서 확인 중이라 하고 내일까지 답변 못 받으면 내일 중간발표 관련 발제 시 문의달라고 답변 받았습니다.

## 🍀 공통 트러블 슈팅

**Comment by eunohhh** [comment]
- Chip 지역선택 노랗게 안되는부분
- 3명까지 선택가능 << 이부분도 옵션으로

- 이제 시간이 없기 때문에 와이어프레임과 상충하는게 있어도, 완전 바껴야 되는거 아니면 일단 만들고보자

- 온보딩시에 4개 / 5개 받는거 안되니까(대비로 하면 안됨 아무튼 db 구조상)
- (칩으로) 3개 받는걸로 해야 될 것 같다고 말씀드리고
- 빠진 내용들(성별, 나이 등도)

### 유연하게 아무튼 funnel 구현해보고 보여드리기(양해는 슬랙으로 구하고)
### 다른 분들 파트도 일단 엔지니어가 판단해서 구현하고 보여드리기

**Comment by Kyara0324** [comment]
- .env.local 파일에 토스 테스트 결제 key 삭제
- tutorial 관련 더미데이터 이미지 webp로 변경

**Comment by eunohhh** [comment]
## 검색

- mvp 까지는
- 검색할 경우에 다 펼쳐져 있고, 스크롤이 내려가게 대처

## 온보딩

- 기술이 허락하는 데까지...

## 노티피케이션

- 페이지로 나눠서
- 5-6가지 여서 데스크탑에서 안허전 할것 같기도...

## 디자이너님 가능시간

- 화 / 목 / 금 오후7시 

## api

- 버디즈 : 오은
- 스토리테이블 : 오은
- 스토리라이크 : 오은
- 트립스 : 김병준
- 트립북마크 : 김병준
- 메시지스 : 김민지
- 노티피케이션 : 김민지

## UT 이후
- 참가 모달 
- 노티피케이션
- 수락
- 강퇴
- 검색 결과 페이지
- 리치텍스트 에디터
- 인공지능 이미지 생성

**Comment by Captain-Kim** [comment]
- 검색 기능 현행 유지 -> UT 이후 최초 기획안 대로 구현
- 온보딩 데이터 현 와이어프레임 칩 5개 선택 -> 3개로 변경 (DB 구조 상)
- [디자이너님] 온보딩에서 받는 정보가 너무 많은 듯 -> [은, 병준] provider 회원가입의 경우 추가 정보를 획득할 수 없어 온보딩 외 받을 수 있는 방법이 없는 상태 -> 현행 유지하면서 사용자의 피로도 개선을 위해 UI 세부 개선
- 디자이너님 가용 시간 : 이번 주 화 목 금
- 알림 페이지(종 모양) : 페이지 이동(디자이너님)
- 인가 : 미들웨어 -> 일요일 / 마이페이지 조건부 렌더링 또는 접근 불가 처리
- UT 이후 구현
  - notifications
  - 채팅 요청 알림 모달 (UT 이전 뱃지로만)
  - 수락/강퇴
  - 검색 결과 페이지
  - 리치텍스트에디터
  - 글쓰기 사진 미선택 시 인공지능 이미지 생성
